pkg/server/webhook.go

```diff
@@ -227,6 +227,41 @@ func addVolumes(target, added []corev1.Volume, basePath string) (patch []patchOp
 	return patch
 }
 
+func addVolumeMounts(target []corev1.Container, addedVolumeMounts []corev1.VolumeMount) (patch []patchOperation) {
+	var value interface{}
+	for containerIndex, container := range target {
+		// for each container in the spec, determine if we want to patch with any volume mounts
+		first := len(container.VolumeMounts) == 0
+		for _, add := range addedVolumeMounts {
+			path := fmt.Sprintf("/spec/containers/%d/volumeMounts", containerIndex)
+			hasKey := false
+			// make sure we dont override any existing volume mounts; we only add, dont replace
+			for _, origVolumeMount := range container.VolumeMounts {
+				if origVolumeMount.Name == add.Name {
```

Comment:
do you want to return an error?

Reply:
Why would I return an error here, as we're consciously checking for matching names to not override an existing volumeMount? I could write a log message saying that the volume mount exists already and that it's not being replaced, but probably that won't be beneficial or what do you think?

Reply:
It depends on what semantics you want.  An `add` of an already existing mount could be interpreted as something out of wack.  In the context of `merge`ing behavior this make more sense to me.

Reply:
this logic is sound to me